### ISSUE TITLE
Option to use anidb for episode information

### ISSUE THREAD

**Issue by SonarrBot** [Sonarr/Sonarr#365]
This is a pretty significant change, so it deserves its own card. ~~We'll need to work with their UDP API, which on first look, is a bit daunting.~~ WE can use the HTTP API which is pretty straight forward.

Searching for new series: http://wiki.anidb.net/w/API#Anime_Titles (We'll want to cache this within services I think)

Updating info: http://wiki.anidb.net/w/HTTP_API_Definition

Client ID: sonarr
Version: 2

We'll also need to consider impact on thexem mappings and other assumptions we make about only using trakt/tvdb. Also the impact on the add series view since seasons aren't a factor.

**Comment by matthewdias** [comment]
thexem has anidb mappings now, which should make this much more doable

**Comment by XVicarious** [comment]
I might be interested in taking this on. But just for reference, we are talking about the separation of anime titles a la AniDB? Like instead of "To Love-Ru" seasons 1-4 plus specials, it would be "To Love-Ru", "To Love-Ru OVA", "Motto To Love-Ru" etc?

**Comment by markus101** [comment]
Each anidb series would be a different series within Sonarr, not separate seasons like TheTVDB does.

All traffic to TheTVDB goes through our proxy, Skyhook, we may want/need to do something similar depending on anidb works exactly, I looked at it once a long time ago, but I've long forgotten how it works and what it offers.

This is definitely a big undertaking with a lot to discuss/decide on as it designed and built, it'd be best to reach out to us on IRC to discuss further.

**Comment by atraholt** [comment]
One option is to use the AnimeLists from ScudLee over on https://github.com/ScudLee/anime-lists

**Comment by xelra** [comment]
The real problem with this request is that there's no value in it for Sonarr's use-case.

Sonarr's purpose is to create tvdb-compliant libraries that work well with apps like Plex and trakt.
Of course you can say that there are anidb scrapers for Plex. But that's actually just running in circles. Since the anidb scraper for Plex makes heavy use of tvdb via ScudLee's list. So then we'd be converting tvdb to anidb in Sonarr and in Plex we'd be converting anidb to tvdb. Or the other way around. However you look at it.
This doesn't add anything to what we have now, but brings along a lot of possible quirks and bugs. Things that need manual intervention and fixing.

It would make sense, if you can get everything on the "appchain" to use anidb. Not just convert, but really use it as the center of getting metadata and usage tracking. I don't see that happening though. For once because of the heavy usage restrictions on their api and also because they simply don't have everything that we need. Like for example high-res posters and backdrops.

**Comment by M0UL** [comment]
i agree with xelra on this.  it make no sense to use the anidb. since everything else is based on the tvdb anyway.

**Comment by MindTooth** [comment]
> i agree with xelra on this. it make no sense to use the anidb. since everything else is based on the tvdb anyway.

That is false.  As there is many shows that don't follow the same order.  An example is Tokyo Ghoul who on TheTVDB has only one show, but is parted on AniDB for each "version/season" of the show.

**Comment by XVicarious** [comment]
That @Birger that was my problem. I ended up just switching to Flexget.
They didn't have support for AniDB either, but the beauty of it was that
its extensible so I ended up writing a few plugins for it. So now Flexget
has (currently) limited support for AniDB.

**Comment by M0UL** [comment]
> 
> 
> > i agree with xelra on this. it make no sense to use the anidb. since everything else is based on the tvdb anyway.
> 
> That is false. As there are many shows that don't follow the same order. An example is Tokyo Ghoul who on TheTVDB is only one show, but is parted on AniDB for each "version/season" of the show.

that is not what i meant with that.  i meant other software like plex and trakt are al using thetvdb based data.

**Comment by MindTooth** [comment]
I use https://github.com/ZeroQI/Hama.bundle to make sure I get metadata from AniDB.  But I will agree that for the most part, people tend to use TheTVDB.

**Comment by XVicarious** [comment]
TheXEM gives mappings for at least most anime to AniDB. I know Sonarr is
pretty integrated into TheTVDB, but considering it's already using a
resource that can be easily used to map to AniDB, its at least worth a look
at, considering TheTVDB doesn't properly represent a fairly popular type of
show.

**Comment by NCLI** [comment]
For people like me, who watch a lot of anime, the TVDB is all but useless. Most shows are released with AniDB naming. It]s just the defacto standard for anime.

**Comment by xelra** [comment]
@NCLI What most people don't seem to understand is that the goal here is to make a tvdb-compliant library. For htpc setups with Plex, Kodi or Emby.

The next argument that people throw in is that there are Anidb based parsers for Plex and Kodi. But that misses the point, because those map back to tvdb via ScudLee's tvdb-anidb list.
So we'd only be running in circles and open ourselves up to lots of bugs and mishaps while accomplishing nothing.

**Comment by NCLI** [comment]
@xelra Fair point. But you must also be able to recognise that, for a non-insignificant part of the userbase, AniDB-integration is a very attractive feature. That you don't want to implement it is understandable, but it's just as understandable that people keep asking for it, I think.

After all, the is no other application quite as good as Sonarr for TV library management. It's hard not to feel that it's a shame it isn't as good as it could be for Japanese content.

**Comment by xelra** [comment]
@NCLI It's not that nobody wants to do pure AniDB parsers, but that it's not possible.
AniDB imposes severe restrictions on its api. And even if we were to mirror it, it still doesn't have everything needed. Especially when it comes to posters and fanart. So the solution would again be an aggregate, which then would be a complete new metadata provider on its own that only receives part of its data from AniDB.

**Comment by aktersnurra95** [comment]
@xelra I'm not sure I'm speaking for everyone but in my case what I'm looking for is the ability to use AniDBs season/episode ordering, which is the one thing that differs completely from TVDB.
I don't really care whether posters and info are grabbed from TVDB as long as my anime can be sorted both in absolute sorting and with second seasons that counts as different shows according to AniDB and the rest of the anime community...
Cheers!

**Comment by aktersnurra95** [comment]
@xelra For example, the absloute sorting works for series that are one long season according to AniDB but split up in about 20 seasons on TVDB, like say One Piece, Sonarr picks up all my episodes since they are all in the same folder on my system, but numbered absolutely which I believe Sonarr supports.

However shows that are comprised of multiple "series" according to AniDB but is only multiple seasons on TVDB can only detect one of the seasons on my system because they are stored in different folders.
Say for example One Punch Man, the second season just aired, but in the anime world its split up into two shows. One punch man and One punch man season 2. Not One punch man season 1 and 2.

The same is true for multiple other shows (like a fourth of all shows).
Look at Food Wars/Shokugeki no Souma, fist season is called just that. But the second is Ni no Sara, third is San no Sara and fourth/third part two is San no Sara: Tootsuki Ressha Hen.
So if I want Sonarr to keep track on the episodes it can only check one folders episodes, and since all shows are season 1 of that show, if I where to add all folders to Sonarr then it would only find 4 duplicates of season 1 since it's Food Wars season 1, Food Wars: Ni no Sara season 1, etc.

Also, I misspoke in my last comment, Absolute sorting alrealy works as I just wrote, but the last part is whats comlpetely broken. Non absolute sorting with different show names.

Cheers!

**Comment by xelra** [comment]
@aktersnurra95 Yes, I and everyone on the Sonarr dev and support team are fully aware of this issue. The way Japanese anime is release, is not compatible with the tv-scene rules. It's not just anime, there are more schemes out there, for example on Korean television that are incompatible.

A few years ago, we came up with a way to map anime release names into tvdb. And suddenly we were able to get anime into the whole tvdb toolchain, with Plex, automated downloads and everything. Because that's what we have tooling for.
And it wasn't the first attempt. I've been on this issue for years of my life.

Unfortunately AniDB is not evolving anymore. And I have no illusions that it ever will. To realize a complete anime toolchain, there's one essential thing missing. And that's a metadata provider similar to tvdb. 
Let's say you were to develop and run a new service called SuperAniDB. You would feed this service with AniDB data and covers and fanart and additionally provide proper and accessible apis. Then you would also need the staff to manually edit issues and maintain the site. And then you would get acceptance in the community, with third-party devs developing the toolchain for Emby, Kodi, Plex, Sonarr, FlexGet, FileBot and everything else.
Then yes, it is possible.
